### PR TITLE
Report the item id that generated the exception

### PR DESCRIPTION
**Pull request by michaelnt**
If the yaml file of an item is edited directly then it can cause an exception when later doorstop commands are run, e.g. publish. This change displays the UID of the item that caused the exception making it easier to fix.

### CODE REVIEW

File: doorstop/core/document.py [hunk 1]
```diff
@@ -193,7 +193,11 @@ def _iter(self, reload=False):
                 else:
                     self._items.append(item)
                     if reload:
-                        item.load(reload=reload)
+                        try:
+                            item.load(reload=reload)
+                        except Exception as e:
+                            print("!! Exception loading %s" % item)
```
Comment (jacebrowning) [inline doorstop/core/document.py]: I think this might be better as: `log.warning("Unable to load: %s", item)`

File: doorstop/core/document.py [hunk 2]
```diff
@@ -193,7 +193,11 @@ def _iter(self, reload=False):
...
+                            raise e
```
Comment (jacebrowning) [inline doorstop/core/document.py]: I believe this can simple be a bare `raise` statement and you can remove `as e` from above.

**Comment by michaelnt** [comment]
Ok, should the print be a log.err?

**Comment by jacebrowning** [comment]
That would work too.

**Comment by michaelnt** [comment]
I think an error probably makes more sense as the program stops running at this point. Elsewhere WARNING is used for missing links, text etc. but the program continues to run.

**Pull request merged**